src/hue.js

```diff
@@ -0,0 +1,24 @@
+import make from "./make";
+
+export const setHue = (color, hueValue) => {
+  if (hueValue === null) return color;
+  const { s, l, a } = color;
+  return make.fromHsl(hueValue, s, l, a);
+}
+
+export const spin = (color, degrees = 0) => {
+  const { h } = color;
+  const newHue = h + degrees;
+
+  if (newHue < 0) {
+    return setHue(color, 360 - Math.abs(newHue));
+  }
+
+  if (newHue > 360) {
+    return setHue(color, newHue - 360);
+  }
+
+  return setHue(color, newHue);
```

Comment:
Not quite sure this would work @prestonvanloon. If you had a base color with a `hue` of `40` and wanted to spin it `-100`:

`40 + (-100) // -60`
`Math.abs(-60) % 360 // 60` <- should be 300